dashboard/src/main/java/com/google/cloud/tools/opensource/cloudbomdashboard/VersionData.java

```diff
@@ -0,0 +1,268 @@
+        templateData.put("lastUpdated", LocalDateTime.now());
+        //Our all-versions dashboard page is a special case
+        if (versions.size() > 1) {
+            templateData.put("staticVersion", "All Versions");
+        } else if (versions.size() == 1) {
+            templateData.put("staticVersion", versions.get(0));
+        }
+        return templateData;
+    }
+
+    private static String latestVersion(Artifact artifact) {
+        if (artifactToLatestVersion.containsKey(artifact)) {
+            return artifactToLatestVersion.get(artifact);
+        }
+        String pomPath = getMetadataURL(artifact);
+        try {
+            URL url = new URL(pomPath);
+            URLConnection conn = url.openConnection();
+            conn.setConnectTimeout(2000);
+            conn.setReadTimeout(2000);
+            Scanner s = new Scanner(conn.getInputStream());
+            while (s.hasNextLine()) {
+                String string = s.nextLine();
+                if (string.contains("<latest>")) {
+                    String version = string.split(">")[1].split("<")[0];
+                    artifactToLatestVersion.put(artifact, version);
+                    return version;
+                }
+            }
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        artifactToLatestVersion.put(artifact, "");
+        return "";
+    }
+
+    private static String updatedTime(Artifact artifact) {
+        if (artifactToTime.containsKey(artifact)) {
+            return artifactToTime.get(artifact);
+        }
+        String groupPath = artifact.getGroupId().replace('.', '/');
+        String metadataPath = DashboardMain.basePath + "/" + groupPath
+                + "/" + artifact.getArtifactId()
+                + "/maven-metadata.xml";
+        try {
+            URL url = new URL(metadataPath);
+            URLConnection conn = url.openConnection();
+            conn.setConnectTimeout(2000);
+            conn.setReadTimeout(2000);
+            Scanner s = new Scanner(conn.getInputStream());
+            while (s.hasNextLine()) {
+                String string = s.nextLine();
+                if (string.contains("<lastUpdated>")) {
+                    DateFormat dateFormat = new SimpleDateFormat("yyyyMMddhhmmss");
+                    DateFormat outputFormat = new SimpleDateFormat("MM-dd-yyyy");
+                    String input = string.split(">")[1].split("<")[0];
+                    Date date = dateFormat.parse(input);
+                    artifactToTime.put(artifact, outputFormat.format(date));
+                    return outputFormat.format(date);
+                }
+            }
+        } catch (IOException | java.text.ParseException e) {
+            e.printStackTrace();
+        }
+        artifactToTime.put(artifact, "");
+        return "";
+    }
+
+    /**
+     * @param key                Key to use when inserting the artifact's associated path into the given map.
```

Comment:
params begin with lower case; i.e. Key --> key
and no period unless there's a complete sentence to follow